db/memtable.h

```diff
@@ -603,7 +603,12 @@ class MemTable {
 
   // Always returns non-null and assumes certain pre-checks are done
   FragmentedRangeTombstoneIterator* NewRangeTombstoneIteratorInternal(
-      const ReadOptions& read_options, SequenceNumber read_seq);
+      SequenceNumber read_seq);
+
+  // The current fragmented range tombstones of this memtable.
+  // Updated when a new DeleteRange() is inserted into the memtable.
+  std::shared_ptr<FragmentedRangeTombstoneList>
+      fragmented_range_tombstone_list_;
```

Comment:
I wonder if we should try ThreadLocalPtr or CoreLocalArray to hold references to the FragmentedRangeTombstoneList. Core-local is usually preferred nowadays. The FragmentedRangeTombstoneList references could be locally refcounted so we only mutate the shared refcount when the list changes.

I found this folly class interesting: https://github.com/facebook/folly/blob/4b930941724f3d2ed44dd9d7e389437dbf26d787/folly/concurrency/CoreCachedSharedPtr.h#L162. It uses shared_ptr for local refcounting (`holder`) of another shared_ptr. We might be able to use CoreLocalArray to do something similar. (I suspect our interface maybe should not be specific to shared_ptr; more details to come on that...)